src/components/Input/DropDown/DropDownGroup.js

```diff
@@ -0,0 +1,271 @@
+import { Provider as DropDownProvider } from "./Context";
+import SelectionProvider from "../../SelectionProvider/Provider";
+import { Consumer as SelectionConsumer } from "../../SelectionProvider/Context";
+
+const ARROWUP = 38;
+const ARROWDOWN = 40;
+const SPACEBAR = 32;
+
+const StyledGroup = styled.label`
+  color: ${colors.onyx.light};
+  width: 100%;
+  height: 44px;
+  background-color: white;
+
+  border-radius: 2px 2px 2px 2px;
+  position: relative;
+  padding: 0px;
+  box-sizing: content-box;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  &.dropDown--border {
+    border: solid 1px ${colors.onyx.muted};
+    text-align: left;
+    &:hover {
+      margin: -1px -1px -1px -1px;
+      border: solid 2px ${colors.azure.base};
+    }
+  }
+
+  &.dropDown--no-border {
+    border: solid 1px ${colors.white.base};
+    text-align: right;
+    justify-content: flex-end;
+  }
+
+  &:focus {
+    margin: -1px -1px -1px -1px;
+    border: solid 2px ${colors.azure.base};
+  }
+
+  &.dropdown--active {
+    margin: 0 0 0 0;
+    border: solid 1px ${colors.onyx.muted};
+    border-radius: 2px 2px 0px 0px;
+    &:hover {
+      margin: 0 0 0 0;
+      border: solid 1px ${colors.onyx.muted};
+    }
+  }
+`;
+
+const StyledChildWrapper = styled.div`
+  display: none;
+  background-color: white;
+  border-radius: 0 0 2px 2px;
+  border: solid 1px ${colors.onyx.muted};
+  padding-top: 4px;
+  padding-bottom: 8px;
+  white-space: nowrap;
+  margin-top: -1px;
+  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
+  min-width: 100%;
+  box-sizing: content-box;
+
+  &.dropDown--clicked {
+    flex-direction: column;
+    position: absolute;
+    overflow-y: auto;
+    flex: 1;
+    display: flex;
+    overflow: auto;
+    height: 135px;
+  }
+`;
+
+const StyledGroupWrapper = styled.div`
+  position: relative;
+  color: grey;
+  width: 100%;
+`;
+
+const StyledChevron = styled(ChevronIcon)`
+  margin-right: 12px;
+
+  &.dropDown__icon--hide {
+    display: none;
+  }
+`;
+
+const StyledSelectedText = styled.div`
+  font-size: 16px;
+  white-space: nowrap;
+
+  &.dropDown--border {
+    margin-left: 15px;
+    width: 60%;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+
+  &.dropDown--no-border {
+    margin-right: 8px;
+    width: 60%;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+  &.dropDown--alignment {
+    margin-right: 32px;
+  }
+`;
+
+const StyledKeyboardProvider = styled(KeyBoardProvider)`
+  min-height: min-content;
+  flex-direction: column;
+  display: flex;
+`;
+
+class DropDownGroup extends React.Component {
+  componentDidMount() {
+    document.addEventListener("click", this.handleOutsideClick);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener("click", this.handleOutsideClick);
+  }
+
+  onClick = () => {
+    this.setState(({ isOpen }) => ({ isOpen: !isOpen }));
+  };
+
+  onKeyDown = e => {
+    if (e.keyCode === ARROWUP || e.keyCode === ARROWDOWN) {
```

Comment:
Should this conditional also check that the `!state.isOpen`  as well?

Reply:
not really because if its open the focus shifts to the children.